data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java

```diff
@@ -82,6 +89,38 @@ private void initialize(final DataProvider dataProvider, final TekuConfiguration
     addCustomErrorPages(configuration);
   }
 
+  private void addHostWhitelistHandler(final TekuConfiguration configuration) {
+    app.before(
+        (ctx) -> {
+          String header = ctx.header("host");
+          Optional<String> optionalHost = getAndValidateHostHeader(header);
+          if (!configuration.getRestApiHostWhitelist().contains("*")
+              && (!optionalHost.isPresent()
+                  || !hostIsInWhitelist(configuration, optionalHost.get()))) {
+            ctx.status(SC_FORBIDDEN);
+            LOG.info("Host not authorized " + optionalHost);
+          }
+        });
+  }
```

Comment:
whitelisting is more of a 'business as usual', i'd leave the log statement as debug.
Also, i can't see any test cases exercising any of this logic...
- does substring get me around the logic (my.host allowing my.)
- does it actually block me if i'm not in the list
- does star let me in
- does my ip / hostname let me in

This may have to be slightly restructured to allow testing to happen cleanly... but it shouldn't be huge, just let me know if you need a hand.